### PR TITLE
icon-9.5.20h

### PR DESCRIPTION
**Pull request by CheyenneWills**
Upstream has switched over from an "infrequent formal release to a
system of continual modifications that are immediately available
publicly through Github".

Going forward upstream will tag the repo and set the version number to
"9.5.yyx" where yy is the last two digits of the year and x is an
alphabetic sequence character within the year (e.g. 9.5.20h).

Create a new ebuild for icon-9.5.20h that supports the new upstream
source and follows upstream versioning.

Replace the use of the patches and use inline sed commands to modify the
Makefiles to follow Gentoo's CFLAGS/LDFLAGS policies.

The changes associated with #669330, #716212 and #732507 are included in
the upstream source.

Bug:https://bugs.gentoo.org/736096

Signed-off-by: Cheyenne Wills <cwills@witznd.net>

### CODE REVIEW

**Comment by thesamesam** [comment]
Can you update the commit message to adhere to GLEP [66](https://www.gentoo.org/glep/glep-0066.html#commit-messages)? (e.g. "dev-lang/icon: bump to 9.5.20h")

**Comment by CheyenneWills** [comment]
Updated commit message subject line: "dev-lang/icon bump to 9.5.20h"

File: dev-lang/icon/icon-9.5.20h.ebuild [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+S="${WORKDIR}/icon-${PV}"
+
+RDEPEND="
+	X? ( x11-libs/libX11:= )"
+DEPEND="
+	${RDEPEND}
+	X? (
+		x11-libs/libXpm
+		x11-libs/libXt
```
Comment (thesamesam) [inline dev-lang/icon/icon-9.5.20h.ebuild:26]: Are these libraries not linked against => needed at runtime?

Apologies if I've asked this before.

File: dev-lang/icon/icon-9.5.20h.ebuild [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	)"
+
+src_prepare() {
+	default
+	# Add CFLAGS and LDFLAGS to Makefiles as needed
+	sed -i -e 's/CFLAGS="$(CFLAGS)"/& LDFLAGS="$(LDFLAGS)"/g' ipl/cfuncs/Makefile
```
Comment (thesamesam) [inline dev-lang/icon/icon-9.5.20h.ebuild:32]: || die on all sed

File: dev-lang/icon/icon-9.5.20h.ebuild (hunk 1 above)
Comment (CheyenneWills) [reply dev-lang/icon/icon-9.5.20h.ebuild:26]: I'm not quite sure.. The source contains a stripped libXpm so the rdepend could simply be ensuring that the libXpm runtime prereqs are there.

File: dev-lang/icon/icon-9.5.20h.ebuild (hunk 2 above)
Comment (CheyenneWills) [reply dev-lang/icon/icon-9.5.20h.ebuild:32]: ACK 
Will do.

**Comment by CheyenneWills** [comment]
Is there anything needed from my end to make this PR move forward?

Thanks

**Comment by thesamesam** [comment]
Merged, but please address:
```
 * QA Notice: Files built without respecting LDFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/lib64/icon/bin/libcfunc.so
```

(I merged anyway because I'm aware of the complexities with this package.)

**Pull request closed without merging**